### PR TITLE
[Enhancement Shaman] Updates

### PR DESCRIPTION
**Pull request by Seriousnes**
### Description

* Malestrom Weapon tracker has been rewritten to provide more accurate tracking
* Added maelstrom usage details table
* Added maelstrom efficiency table

### Testing
- Test report URL: `/report/WcFTvqQHZJhKwNzp/19-Mythic+Scalecommander+Sarkareth+-+Wipe+16+(5:39)/Seriousnes/standard/statistics`
- Screenshot(s):

Usage details
image

Efficiency
image

### CODE REVIEW

File: src/analysis/retail/shaman/enhancement/modules/normalizers/EventLinkNormalizer.tsx
Comment (emallson) [inline src/analysis/retail/shaman/enhancement/modules/normalizers/EventLinkNormalizer.tsx]: typo

```suggestion
const maelstromWeaponSpenderLink: EventLink = {
```

File: src/analysis/retail/shaman/enhancement/modules/dragonflight/Tier30.tsx
Comment (emallson) [inline src/analysis/retail/shaman/enhancement/modules/dragonflight/Tier30.tsx]: just as a note: the `key` prop is only needed when using arrays as JSX. when you've spelled it out like this the `key` prop doesn't add anything. not something that needs fixing, but a note

File: src/analysis/retail/shaman/enhancement/modules/resourcetracker/MaelstromWeaponSpenders.tsx
Comment (emallson) [inline src/analysis/retail/shaman/enhancement/modules/resourcetracker/MaelstromWeaponSpenders.tsx]: note that we have the `GetRelatedEvents` helper for cases like this (it is a little more awkward for cases with a single event like PW LB)

```suggestion
      const damageEvents: DamageEvent[] = GetRelatedEvents(primordialWaveLightningBolt, LIGHTNING_BOLT_PRIMORDIAL_WAVE_LINK) as DamageEvent[];
```

File: src/analysis/retail/shaman/enhancement/modules/resourcetracker/MaelstromWeaponTracker.tsx
Comment (emallson) [inline src/analysis/retail/shaman/enhancement/modules/resourcetracker/MaelstromWeaponTracker.tsx:38]: minor style thing: setting the class name here can help in debugging to make it clearer what type an object is. not required to merge, though

**Review by emallson — APPROVED** [review]
sorry for the late review!

some minor comments but nothing that actually *requires* intervention before merging, so i'll merge sometime tomorrow unless i hear something from you about wanting to address any comments

File: src/analysis/retail/shaman/enhancement/modules/dragonflight/Tier30.tsx
Comment (Seriousnes) [reply src/analysis/retail/shaman/enhancement/modules/dragonflight/Tier30.tsx]: I'll remove them, I think I was a bit overzealous when trying to fix the missing key properties I just added it to every `<li>` I've used in the entire enhancement module

File: src/analysis/retail/shaman/enhancement/modules/resourcetracker/MaelstromWeaponSpenders.tsx
Comment (Seriousnes) [reply src/analysis/retail/shaman/enhancement/modules/resourcetracker/MaelstromWeaponSpenders.tsx]: Done

File: src/analysis/retail/shaman/enhancement/modules/resourcetracker/MaelstromWeaponTracker.tsx
Comment (Seriousnes) [reply src/analysis/retail/shaman/enhancement/modules/resourcetracker/MaelstromWeaponTracker.tsx:38]: I'll keep this in mind for next time

**Review by emallson — APPROVED** [review]

**Pull request merged**
